types/ember__routing/-private/transition.d.ts

```diff
@@ -42,7 +42,7 @@ export default interface Transition<T = any> extends Partial<Promise<T>> {
     catch<TResult = never>(
         onRejected?: (reason: any) => TResult | PromiseLike<TResult>,
         label?: string,
-    ): Promise<TResult>;
+    ): Promise<TResult | T>;
```

Comment:
This is a separate bugfix I noticed that brings it more inline with `Promise` and matches the actual behavior.